src/mono/sample/wasm/browser-threads-minimal/Program.cs

```diff
@@ -146,6 +238,14 @@ public static async Task<int> RunBackgroundTaskRunCompute()
             return rs[0];
         }
 
+        [JSExport]
+        internal static void GCCollect()
+        {
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
```

Comment:
We talked offline about

```csharp
namespace System;

public partial class GC {
   public static Task AwaitForPendingFinalizers();
}
```

That way the JS code could await in the JS event loop for the finalizer thread to run and clean up the webworkers for threads between tests.  (the synchronous wait is probably not giving it a chance to do the message passing that will ensure that the webworker returns to emscripten's worker pool before the next test runs).

This can wait until a follow-up PR.